.gitignore

```diff
@@ -1,6 +1,7 @@
 # Project Files
 *.tmp
 
+config.json
 index-test.js
```

Comment:
Will do, my original idea for this was just to make it easier to update, the credentials thing was something I thought of after the fact.

The original idea being that you could use git to distribute updates for the common user that's not going to change anything other than the config so the only thing they would have to do at that point is pull down the changes from master making any update easy to do.

Reply:
I hadn't thought of that when replying originally, but I don't think it changes anything.

Even after adding credentials `git pull` should still work in most cases. The only cases that I can think of where merge conflicts occur are when a significant change has been made to the config or a default value has been changed, which is quite rare and when does occur is well documented on Discord, etc.. The benefit of allowing git to update configs in most cases it will be handled fine and new plugins/options will be added without users having to manually copy them to their config. Having said that, I don't experience updating from a user perspective much, so I not sure whether others feel differently?

Reply:
my understanding is any tracked file will be overwritten via `git pull` by the the file from the repo if it was edited but not committed/stashed, the idea being to not track the config file in order to preserve at least the basic settings for a server.

I'm almost thinking it might be best to split it into two files server_config and plugins_config so that the things that never really change can be in a static untracked file (server_config) and the plugins can be overwritten with the newest recommended default options, I'd think this would work for 90% of users and also provide a command to give to users for a simple way to update (`git pull`).